### ISSUE TITLE
breaking change to EDTF masked precision syntax in edtf-ruby v3.1

### ISSUE THREAD

**Issue by thatbudakguy** [sul-dlss/mods#54]
we've started to see [CI failures in gems](https://github.com/sul-dlss/stanford-mods/pull/131) like `stanford-mods` and `mods` that depend on `ruby-edtf` that appear to be caused by [changes in v3.1 of that gem](https://github.com/inukshuk/edtf-ruby/pull/36) (link is to discussion on reason for the changes).

the problems all seem to be related to the interpretation of decade- and century-precision EDTF dates as intervals using the so-called "masked precision syntax". in the new version of `ruby-edtf`:
```ruby
> Date.edtf('190x').class
EDTF::Decade # correctly parsed as decade
> Date.edtf('190X').class
Date # not a decade
> Date.edtf('190u').class
Date # not a decade
```
in other words, the only supported indicator of decade or century precision is lowercase `x`, not `X` or `u`. I'm not familiar with the spec and am not sure whether this change is warranted. it certainly seems like we have a lot of tests presuming it's OK to indicate decade and century precision in other ways — for example:
https://github.com/sul-dlss/mods/blob/6f4fc12b66736f3ddbe97506cbbc03f39694ac6e/spec/lib/date_spec.rb#L204-L207

for now I'm going to pin `ruby-edtf` so that we don't pick up this change anymore, but would like to hear from @arcadiafalcone or anybody who knows more than me about what might be going on here. are the `ruby-edtf` folks wrong about the spec?

**Comment by arcadiafalcone** [comment]
According to the LC documentation for the final EDTF specification (https://www.loc.gov/standards/datetime/), the character for unspecified digits should be upper case "X", and lower case "u" was removed when the draft version was finalized. I don't see anything about a lower case "x".

**Comment by cbeer** [comment]
Looks like lower-case `x` got dropped in the final spec:
> Masked precision is eliminated;

**Comment by thatbudakguy** [comment]
The draft spec from 2012 link is broken, but apparently it was around [in 2019 and so can be wayback'd](https://web.archive.org/web/20190429153902/http://www.loc.gov/standards/datetime/pre-submission.html#maskedprecision).

> In a four-digit date string, 'x' may replace the last digit, or 'xx' the last two digits.
> 
> Examples
> 
>     196x
>     A date during the 1960s
>     19xx
>     A date during the 1900s
> 
> '196x' and '19xx ' have decade precision and century precision respectively.
> 
> Note the difference in semantics between 'x' and 'u'. '196x' has decade precision while '196u' has year precision. Both represent an unspecified year during the 1960s, but for 196x the year is not supplied because it is known only with decade precision. In contrast, for 196u the year is not supplied for reasons that are not specified but there is some expectation (though no guarantee) that the year may be supplied later; for 196x there is no such expectation.

So it seems that ditching the draft spec is indeed a breaking change: `x`, or masked precision, no longer exists. Do we have real records using this syntax? The tests seem to suggest we do.

**Comment by cbeer** [comment]
I wonder if we have real records, or if it's just our munging that's wrong e.g. https://github.com/sul-dlss/mods/blob/master/lib/mods/date.rb#L186.. I think infra has ways to check on date formats

**Comment by arcadiafalcone** [comment]
@mjgiarlo Does the last clean EDTF date validation report we ran use the edtf-ruby v3.1 specs?

**Comment by cbeer** [comment]
I think the dates will parse the same, but the meaning of `"x"` and `"X"` have changed. Do you have a good way to find EDTF dates with `x`?

**Comment by mjgiarlo** [comment]
@cbeer Not currently, but we could whip up a report to pull that data out. To write and run the report, that's probably 1-2 dev-hours, so no big deal?

**Comment by arcadiafalcone** [comment]
@mjgiarlo I'll write up a ticket. @cbeer To confirm, it's specifically EDTF dates with lower case `x` that you're interested in?

**Comment by arcadiafalcone** [comment]
@mjgiarlo Report request ticket: https://github.com/sul-dlss/dor-services-app/issues/4143
